server/src/UnifiedDataStore.cc

```diff
@@ -379,14 +379,9 @@ void UnifiedDataStore::reset(void)
 
 UnifiedDataStore *UnifiedDataStore::getInstance(void)
 {
-	if (Impl::instance)
-		return Impl::instance;
-
-	Impl::mutex.lock();
+	AutoMutex lock(&Impl::getInstanceMutex);
```

Comment:
I don't want mix different way for the same purpose in a same file.
AutoMutex is already used at another place in the same file.

We should do it in another PR.